spec/services/publish_reporting_service_spec.rb

```diff
@@ -0,0 +1,120 @@
+require "rails_helper"
+
+describe PublishReportingService do
+  let(:expected) do
+    {
+      users: {
+        total: {
+          all: 666,
+          active_users: 600,
+          non_active_users: 66,
+          active_users_last_30_days: 60,
+        },
+      },
+      providers: {
+        total: {
+          all: 1000,
+          providers_with_non_active_users: 990,
+          providers_with_active_users: 10,
+        },
+        with_1_active_users: 1,
+        with_2_active_users: 2,
+        with_3_active_users: 0,
+        with_4_active_users: 4,
+        with_more_than_5_active_users: 6,
+      },
+      courses: {
+        total_updated_last_30_days: 100,
+        updated_non_findable_last_30_days: 40,
+        updated_findable_last_30_days: 60,
+        updated_open_courses_last_30_days: 35,
+        updated_closed_courses_last_30_days: 25,
+        created_last_30_days: 10,
+      },
+    }
+  end
+
+  let(:recruitment_cycle_scope) do
+    instance_double(RecruitmentCycle)
+  end
+  let(:courses_scope) { class_double(Course) }
```

Comment:
```suggestion
  let(:courses_scope) { Course::ActiveRecord_Relation }
```

Reply:
`uninitialized constant Course::ActiveRecord_Relation`

Reply:
Remember we were debating just using `Course.all`? Just use that.